index.rst

```diff
@@ -105,10 +123,9 @@ This class is a subclass of ``dmlc::Parameter``.
 .. parsed-literal::
 
   {
-    "base_score" : *string*,
-    "num_feature" : *string*,
-    "num_class" : *string*,
-    "contain_eval_metrics" : *string*
+    "base_score" : *floating-point*,
+    "num_feature" : *64-bit integer*,
+    "num_class" : *integer*
```

Comment:
Is it needed? Isn't the number of classes the same as the number of models in the file (a model for a column of one-hot encoded classes), which is the same as `num_roots` (which is also proposed to be eliminated)?

Reply:
>No, the number of trees = num_class times num_iteration.

Under a model I meant an ensemble of trees, a predictor for any single column resulting from one-hot encoding.

>Num_roots is a separate feature that’s proposed for deprecation.

Should we remove it then? What was the initial rationale to add it?

Reply:
num_class has nothing to do with one hot encoding. It is used for multiclass classification, where the target class is more than two

Reply:
Actually, you are right, if you meant to refer to one-hot encoded target label

Reply:
@RAMitchell I recall you were trying to deprecate num_roots. What do you think?